.github/scripts/windows-agent-compile.ps1

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright 2024 Centreon
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+# For more information : contact@centreon.com
+#
+
+Write-Host "Work in" $pwd.ToString()
+
+[System.Environment]::SetEnvironmentVariable("AWS_EC2_METADATA_DISABLED","true")
+
+Write-Host $env:VCPKG_BINARY_SOURCES
+
+$current_dir = $pwd.ToString()
+
+#get cache from s3
+$files_to_hash= "vcpkg.json", "custom-triplets\x64-windows.cmake", "CMakeLists.txt", "CMakeListsWindows.txt"
+$files_content= Get-Content -Path $files_to_hash -Raw
+$stringAsStream = [System.IO.MemoryStream]::new()
+$writer = [System.IO.StreamWriter]::new($stringAsStream)
+$writer.write($files_content -join " ")
+$writer.Flush()
+$stringAsStream.Position = 0
+$vcpkg_hash=Get-FileHash -InputStream $stringAsStream -Algorithm SHA256 | Select-Object Hash
+$file_name = "windows-agent-vcpkg-dependencies-cache-" + $vcpkg_hash.Hash
+$file_name_extension = "${file_name}.7z"
```

Comment:
Be careful to the coding style.
hash= toto
hash = titi
hash=tata

I'd prefer the second one.